src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2810,13 +2827,24 @@ internal void CheckAndReportRequiresUnreferencedCode (MethodDefinition method)
 			if (ShouldSuppressAnalysisWarningsForRequiresUnreferencedCode ())
 				return;
 
-			if (Annotations.TryGetLinkerAttribute (method, out RequiresUnreferencedCodeAttribute requiresUnreferencedCode)) {
-				string formatString = SharedStrings.RequiresUnreferencedCodeMessage;
-				string arg1 = MessageFormat.FormatRequiresAttributeMessageArg (requiresUnreferencedCode.Message);
-				string arg2 = MessageFormat.FormatRequiresAttributeUrlArg (requiresUnreferencedCode.Url);
-				string message = string.Format (formatString, method.GetDisplayName (), arg1, arg2);
-				_context.LogWarning (message, 2026, currentOrigin, MessageSubCategory.TrimAnalysis);
+			if (method.IsStatic || method.IsConstructor) {
+				if (Annotations.TryGetEffectiveRequiresUnreferencedCodeAttributeOnType (method.DeclaringType, out RequiresUnreferencedCodeAttribute requiresUnreferencedCodeOnTypeHierarchy) && requiresUnreferencedCodeOnTypeHierarchy != null) {
+					ReportRequiresUnreferencedCode (method.GetDisplayName (), requiresUnreferencedCodeOnTypeHierarchy, currentOrigin);
+					return;
+				}
 			}
+
+			if (Annotations.TryGetLinkerAttribute (method, out RequiresUnreferencedCodeAttribute requiresUnreferencedCode))
```

Comment:
Can the `out` parameter be `null` here?

Reply:
Only if we return false in which case we will not execute the code following. Otherwise, the LinkerAttributes cache when gets generated checks for the creation of the right attribute with the right parameters.

Reply:
https://github.com/mono/linker/blob/2174dc1bf8217ab8f5b77b0fcc0ab0cbf1973a85/src/linker/Linker/LinkerAttributesInformation.cs#L34
https://github.com/mono/linker/blob/2174dc1bf8217ab8f5b77b0fcc0ab0cbf1973a85/src/linker/Linker/LinkerAttributesInformation.cs#L88

Reply:
Can we `#enable nullable` for these sections so we have proper tracking? If this can only return `null` when false, then I think the null check on line 2831 is always true

Reply:
Aghh, that was some remaining code of a previous implementation, deleted the check